signac/contrib/project.py

```diff
@@ -1526,11 +1518,42 @@ def init_project(cls, name, root=None, workspace=None, make_dir=True):
             configuration.
 
         """
+        if args:
+            if len(args) == 1:
+                warnings.warn(
+                    "Project names were removed in signac 2.0. If you intended to call "
+                    "`init_project` with a root directory, please provided it as a keyword "
+                    f"argument: `init_project(root={args[0]})`. If your project name "
+                    "contains important information, consider storing it in the project document "
+                    "instead.",
+                    FutureWarning,
+                )
+            else:
+                # Match the usual error from misusing keyword-only args.
+                num_args = len(args)
+                raise TypeError(
+                    f"init_project() takes 0 positional arguments but {num_args} "
+                    f"{'was' if num_args == 1 else 'were'} given"
+                )
+        if kwargs:
+            if len(kwargs) == 1 and "name" in kwargs:
```

Comment:
This should check `"name" in kwargs` by popping it with a default value, and then raise a `TypeError` if other kwargs remain. Otherwise it's possible to pass `name` along other invalid keyword arguments and get an error message about `name`, when `name` would otherwise be supported if the other invalid keyword arguments were removed.
```suggestion
            if kwargs.pop("name", None):
```

Reply:
Or maybe switch this logic around, so it can issue the error _before_ potentially issuing a warning. It would look like:
```python
name = kwargs.pop("name", None)
if kwargs:
    raise TypeError(...)
if name:
    warnings.warn(...)
```